src/_pytest/main.py

```diff
@@ -78,6 +78,11 @@ def pytest_addoption(parser: Parser) -> None:
         type="args",
         default=[],
     )
+    parser.addini(
+        "discover_imports",
+        "Whether to discover tests in imported modules outside `testpaths`",
+        default=False,
```

Comment:
We need to start with a default of true for backwards compatibility 

Alternatively none with a informative warning

Reply:
If it helps, I'd be happy to argue that the current behavior was a substantial surprise to me, so that defaulting to False would count as removing a bug.

Reply:
> If it helps, I'd be happy to argue that the current behavior was a substantial surprise to me, so that defaulting to False would count as removing a bug.

I understand where you are coming from and I agree this would be the reasonable default if pytest was being born today... but unfortunately in this case I suspect there are test suites which rely on this behavior, so we really should be conservative here.